tests/test_objects.py

```diff
@@ -191,16 +192,6 @@ def test_explicit_provider():
     assert sg3.provider == "aws.london"
 
 
-def test_duplicate_key():
-    # ordering should ensure it's always <firstCreatedKey>, <secondCreatedKey>
-    # so let's ensure that works, even if say the value names are backwards-sorted
-    key2 = DuplicateKey("mysql")
-    key1 = DuplicateKey("mysql")
-    encoded = json.dumps({key1: {"user": "wyatt1"}, key2: {"user": "wyatt2"}}, sort_keys=True)
-    desired = '{"mysql": {"user": "wyatt2"}, "mysql": {"user": "wyatt1"}}'
-    assert encoded == desired
```

Comment:
This test is no longer valid.

Previously we did indeed have deterministic ordering because of our monotonically incrementing ID, however this led the collision case that prompted this PR in the first place.

By moving to the Python recommended method of `hash`ing a tuple of the unique parts of our `DuplicateKey` combined with the fact that:

> Note By default, the `__hash__()` values of `str` and `bytes` objects are “salted” with an unpredictable random value. Although they remain constant within an individual Python process, they are not predictable between repeated invocations of Python.

It means that we specifically do NOT have determinism anymore.